custom_components/ecodevices/ecodevicesapi/__init__.py

```diff
@@ -19,29 +19,48 @@ def host(self):
     def _request(self, params):
 
         if self._username is not None and self._password is not None:
-            r = requests.get(self._api_url, params=params, auth=(self._username, self._password), timeout=self._timeout)
+            r = requests.get(
+                self._api_url,
+                params=params,
+                auth=(self._username, self._password),
+                timeout=self._timeout
+            )
         else:
-            r = requests.get(self._api_url, params=params, timeout=self._timeout)
+            r = requests.get(
+                self._api_url,
+                params=params,
+                timeout=self._timeout
+            )
         r.raise_for_status()
         content = r.json()
         product = content.get("product", None)
         if product == "Eco-devices":
             return content
         else:
             raise Exception(
-                "Eco-Devices api request error, url: %s`r%s", r.request.url, content,
+                "Eco-Devices api request error, url: %s`r%s",
+                r.request.url,
+                content,
             )
 
     def ping(self) -> bool:
         try:
             self._request({"cmd": 10})
             return True
-        except:
+        except Exception:
```

Comment:
I'm not entirely sure for this one; my python is "rusty".

Reply:
i'm not sure, I used this : https://wiki.python.org/moin/HandlingExceptions
```import sys
try:
  untrusted.execute()
except: # catch *all* exceptions
  e = sys.exc_info()[0]
  write_to_page( "<p>Error: %s</p>" % e )```

Reply:
OK, so I guess the best way would have been to exclude that one for checks; or to do differently (seem it is what you did on your rewrite branch).